libs/langchain/langchain/agents/agent_toolkits/conversational_retrieval/openai_functions.py

```diff
@@ -0,0 +1,80 @@
+from typing import List, Optional
+
+from langchain.agents.agent import AgentExecutor
+from langchain.agents.openai_functions_agent.agent_token_buffer_memory import (
+    AgentTokenBufferMemory,
+)
+from langchain.agents.openai_functions_agent.base import OpenAIFunctionsAgent
+from langchain.chat_models.openai import ChatOpenAI
+from langchain.memory.token_buffer import ConversationTokenBufferMemory
+from langchain.prompts.chat import MessagesPlaceholder
+from langchain.schema.language_model import BaseLanguageModel
+from langchain.schema.memory import BaseMemory
+from langchain.schema.messages import SystemMessage
+from langchain.tools.base import BaseTool
+
+
+def _get_default_system_message() -> SystemMessage:
+    return SystemMessage(
+        content=(
+            "Do your best to answer the questions. "
+            "Feel free to use any tools available to look up "
+            "relevant information, only if neccessary"
+        )
+    )
+
+
+def create_conversational_retrieval_agent(
+    llm: BaseLanguageModel,
+    tools: List[BaseTool],
```

Comment:
could be nice if this took something like instead/as well
```python
retrievers: Sequence[Tuple[Retriever, str, str]]
```

Reply:
eh i think thats actually more confusing